packages/core/src/shared/utilities/textDocumentUtilities.ts

```diff
@@ -165,3 +165,22 @@ export async function createTempFileForDiff(
     await applyChanges(doc, range, code)
     return tempFileUri
 }
+
+/**
+ * Indents the given code based on the current document's indentation at the selection start.
+ *
+ * @param {any} message - The message object containing the code.
+ * @param {vscode.TextDocument} doc - The VSCode document where the code is applied.
+ * @param {vscode.Selection} selection - The selection range in the document.
+ * @returns {string} - The processed code to be applied to the document.
```

Comment:
Typescript already has the types in the code itself, and any code analyzer (LSP, vscode) will pick this up.
```suggestion
 * @param message The message object containing the code.
 * @param doc The VSCode document where the code is applied.
 * @param selection The selection range in the document.
 * @returns The processed code to be applied to the document.
```